### PR TITLE
Stoplight Indicator Hotfix

### PR DESCRIPTION
**Pull request by DrMaxis**
## Description

This issue came to attention on Feb. 05, 2021. 
Location: Bid -> Pricing

Basis: The stoplight indicator is returning the wrong results for line items that are typed as `labor`.

Evidence: When selecting a single line item (`Racking - Labor - PB3 - L3`) and mapping its data and stoplight results on a spreadsheet, we see differences between the returned values and the values calculated on the spreadsheet. 

Values on the spreadsheet that should match the line item's values

chrome_h2pblb93Es

Values being returned by the UI

Screenshot at Feb 08 10-49-28

Here we can see that the `line_items_value` does not match the spreadsheets, this is causing the indicator to show inaccurate data. 

Solution:

- [x] Add a check to determine if a line item is typed as `labor` or `cost` and use the respected value during stoplight indicator calculations. 
- [x] If a line item is typed as `labor` and the line item's wage & burden is `> 0` &  ` !== undefined`  use `labor_hours` for the value of the line item.

After some discussion, we concluded that  **If a `line item` is typed as `labor_hours`  whose `wage` and `burden` values are greater than zero and not undefined, use its `labor_hours` value for stoplight calculations. This value is used when the value of the `line_item` is compared against the `weightedNormalValues`  to determine its stoplight range.**

### Notes 
-  The values for element number `2` and `5` on the spreadsheet are wrong because they are using `70%` and `30%` while the indicator uses `75%` and `25%`.

- A line item's wage is determined by the location and the labor rate. These are fields that can be modified on the Bid->Fields page. When a line item's location is empty or not set, the wage is set to 0 or undefined.

### CODE REVIEW

File: src/domain/LineItem.js [hunk 1]
```diff
@@ -1681,6 +1694,14 @@ export default class LineItem extends BidEntity {
...
     if (this.getWeightedNormalValues()) {
       weightedNormalValues = this.getWeightedNormalValues();
+      console.log({
+        line_item_title: this.title,
+        weighted_normalValues: weightedNormalValues,
+        predicted_value: this.getPredictedValue(),
+        watts: this.bid._data.watts,
+        line_item_value: this.getValue(),
+        line_item_wage: this._getWageValue(),
+      });
```
Comment (rreynier) [inline src/domain/LineItem.js:1704]: I assume we wanna remove this?

File: src/domain/LineItem.js [hunk 2]
```diff
@@ -1668,6 +1668,19 @@ export default class LineItem extends BidEntity {
...
+   * @returns {number}
+   */
+  getValue () {
+
+    if(this.isLabor()) {
+      return (this._getWageValue() === 0 || typeof this._getWageValue() == 'undefined') ? this._getCostValue() : this._getLaborHoursValue();
+    } else {
+      return this._getCostValue();
+    }
+  }
```
Comment (rreynier) [inline src/domain/LineItem.js:1682]: Could write a quick test for this.

**Review by rreynier — APPROVED** [review]
Few notes, but good to merge once resolved!

File: src/domain/LineItem.js (hunk 1 above)
Comment (DrMaxis) [reply src/domain/LineItem.js:1704]: Yessir. This will definitely be removed.

File: src/domain/LineItem.js (hunk 2 above)
Comment (DrMaxis) [reply src/domain/LineItem.js:1686]: Definitely. 

Planning on writing the test immediately after the merge of these changes the reason being that I would like @connorenglish to test the indicator with his data and provide feedback and make changes (if any).

 I have planned to extend our current `stoplight indicator test` into two tests named `stoplight cost line item test` and `stoplight labor line item test`. 

 For the most part, the tests are the same, but the labor line item test will have a different prediction model which ill map onto a spreadsheet to create our base values. The prediction model will be from line item: `Line Item 2.1` in the `StoplightTestProject` which has a predicted value based on `labor_hours`.

I need to map out this prediction model onto the spreadsheet and create a base of values to compare against the SDKs returned values.

**Currently, this change does not alter the stoplight indicators calculations and the current test for the stoplight passes with these changes.**

**Review by rreynier — APPROVED** [review]
:+1:

**Pull request merged**